test/e2e/network_segmentation_services.go

```diff
@@ -159,14 +158,12 @@ var _ = Describe("Network Segmentation: services", func() {
 				// UDN -> UDN
 				By("Connect to the UDN service cluster IP from the UDN client pod on the same node")
 				checkConnectionToClusterIPs(f, udnClientPod, udnService, udnServerPod.Name)
+				By("Connect to the UDN service nodePort on all 3 nodes from the UDN client pod")
 				checkConnectionToNodePort(f, udnClientPod, udnService, &nodes.Items[0], "endpoint node", udnServerPod.Name)
 				// FIXME(dceara): Remove this check when Local Gateway external->service support is implemented.
 				if !IsGatewayModeLocal() {
-					// FIXME(kyrtapz): Remove once l2 external->svc is fixed. Client node is nodes.Items[0]
-					if netConfigParams.topology != types.Layer2Topology {
-						checkConnectionToNodePort(f, udnClientPod, udnService, &nodes.Items[1], "other node", udnServerPod.Name)
-						checkConnectionToNodePort(f, udnClientPod, udnService, &nodes.Items[2], "other node", udnServerPod.Name)
-					}
+					checkConnectionToNodePort(f, udnClientPod, udnService, &nodes.Items[1], "other node", udnServerPod.Name)
+					checkConnectionToNodePort(f, udnClientPod, udnService, &nodes.Items[2], "other node", udnServerPod.Name)
```

Comment:
This is not going to exercise the remote port since it's using the remote pod's LSP.

Reply:
We can ignore this, since e2e is testing it within a pod, so the remote LSPs are exercised.

Reply:
The remote LSP are not exercised even if the client is at the pod since the OVN LBs convert the nodeport into pod's IP, so traffic goes over the pod's remote LSP.

Reply:
no there is a specific test that exercises the remote port LRP, I tested it using that test :)

Reply:
Righ right, my bad I have just take into account that the pod has to answer back using the remote node IP, and for that we need the remote ports... we can totally disregard my comment

Reply:
But then how can be possible that test are passing without local ports having the tunnel key ?

Reply:
exactly.. hehe I think there is a slight race somewhere .... let's just fix the code to be correct and see